indexing-service/src/test/java/org/apache/druid/indexing/common/task/CachingLocalSegmentAllocatorTest.java

```diff
@@ -0,0 +1,130 @@
+  private static final String DATASOURCE = "DATASOURCE";
+  private static final Interval INTERVAL = Intervals.utc(0, 1000);
+  private static final String VERSION = "version";
+  private static final String DIMENSION = "dim";
+  private static final List<String> PARTITION_DIMENSIONS = ImmutableList.of(DIMENSION);
+  private static final int NUM_PARTITONS = 1;
+  private static final ShardSpecFactory SHARD_SPEC_FACTORY = new HashBasedNumberedShardSpecFactory(
+      PARTITION_DIMENSIONS,
+      NUM_PARTITONS
+  );
+  private static final Map<Interval, Pair<ShardSpecFactory, Integer>> ALLOCATE_SPEC = ImmutableMap.of(
+      INTERVAL, Pair.of(SHARD_SPEC_FACTORY, NUM_PARTITONS)
+  );
+
+  private InputRow row;
+
+  private CachingLocalSegmentAllocator target;
+
+  @Before
+  public void setUp() throws IOException
+  {
+    TaskToolbox toolbox = createToolbox();
+    row = createInputRow();
+
+    target = new CachingLocalSegmentAllocator(toolbox, TASK_ID, DATASOURCE, ALLOCATE_SPEC);
+  }
+
+  @Test
+  public void test_getSequenceName_forIntervalAndRow_shouldUseISOFormatAndPartitionNumForRow()
+  {
+    String sequenceName = target.getSequenceName(INTERVAL, row);
+    String expectedSequenceName = StringUtils.format("%s_%s_%d", TASK_ID, INTERVAL, 0);
+    Assert.assertEquals(expectedSequenceName, sequenceName);
+  }
+
+  private static TaskToolbox createToolbox()
+  {
+    TaskToolbox toolbox = EasyMock.mock(TaskToolbox.class);
+    EasyMock.expect(toolbox.getTaskActionClient()).andStubReturn(createTaskActionClient());
+    EasyMock.expect(toolbox.getObjectMapper()).andStubReturn(OBJECT_MAPPER);
+    EasyMock.replay(toolbox);
+    return toolbox;
+  }
+
+  private static TaskActionClient createTaskActionClient()
+  {
+    List<TaskLock> taskLocks = Collections.singletonList(createTaskLock());
+
+    try {
+      TaskActionClient taskActionClient = EasyMock.mock(TaskActionClient.class);
+      EasyMock.expect(taskActionClient.submit(EasyMock.anyObject(LockListAction.class))).andStubReturn(taskLocks);
+      EasyMock.replay(taskActionClient);
+      return taskActionClient;
+    }
+    catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  private static TaskLock createTaskLock()
+  {
+    TaskLock taskLock = EasyMock.mock(TaskLock.class);
+    EasyMock.expect(taskLock.getInterval()).andStubReturn(INTERVAL);
+    EasyMock.expect(taskLock.getVersion()).andStubReturn(VERSION);
+    EasyMock.replay(taskLock);
+    return taskLock;
+  }
+
+  private static InputRow createInputRow()
+  {
+    long timestamp = INTERVAL.getStartMillis();
+    InputRow inputRow = EasyMock.mock(InputRow.class);
+    EasyMock.expect(inputRow.getTimestamp()).andStubReturn(DateTimes.utc(timestamp));
+    EasyMock.expect(inputRow.getTimestampFromEpoch()).andStubReturn(timestamp);
+    EasyMock.expect(inputRow.getDimension(DIMENSION)).andStubReturn(Collections.singletonList(DIMENSION));
+    EasyMock.replay(inputRow);
+    return inputRow;
+  }
```

Comment:
@ccaominh I copied most these helper initialization functions from your DefaultCaching...AllocatorTest in the refactoring PR

If you want I can consolidate them after your change is merged

Reply:
Yeah, probably good to consolidate.